src/test/java/com/kodilla/ecommercee/entity/ProductEntityTest.java

```diff
@@ -1,7 +1,113 @@
 package com.kodilla.ecommercee.entity;
 
+import com.kodilla.ecommercee.repository.GroupRepository;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
 import static org.junit.Assert.*;
 
+@RunWith(SpringRunner.class)
+@SpringBootTest
 public class ProductEntityTest {
+    private ProductGroupEntity group;
+    private ProductEntity productEntity;
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Before
+    public void createSampleDataInTheDatabase() {
+        ProductGroupEntity groupEntity = new ProductGroupEntity();
+        groupEntity.setGroupName("test group name 1");
+        groupRepository.save(groupEntity);
+        ProductEntity product = new ProductEntity("test name 1", new BigDecimal("1.11"), "test description 1");
+        product.setProductGroupEntity(groupEntity);
+        groupEntity.getProducts().add(product);
+        group = groupRepository.save(groupEntity);
+        productEntity = group.getProducts().get(0);
+        ProductEntity product2 = new ProductEntity("test name 2", new BigDecimal("2.22"), "test description 2");
+        productRepository.save(product2);
+    }
+
+    @After
+    public void cleanUpDatabase() {
+        groupRepository.deleteById(group.getId());
+    }
+
+    @Test
+    public void testCreateProductAndFind() {
+        //Given
+        ProductGroupEntity groupEntity = new ProductGroupEntity();
+        groupEntity.setGroupName("test group name 1");
+        Long groupId = groupRepository.save(groupEntity).getId();
+        ProductEntity productEntity = new ProductEntity("test name 1", new BigDecimal("1.11"), "test description 1");
+        productEntity.setProductGroupEntity(groupEntity);
+        groupEntity.getProducts().add(productEntity);
+        //When
+        Long id = productRepository.save(productEntity).getId();
+        Optional<ProductEntity> result = productRepository.findById(id);
+        //Then
+        assertTrue(result.isPresent());
+        assertEquals("test name 1", result.get().getName());
+        assertEquals(new BigDecimal("1.11"), result.get().getPrice());
+        assertEquals("test description 1", result.get().getDescription());
+        assertEquals(groupId, result.get().getProductGroupEntity().getId());
+        //CleanUp
+        groupRepository.deleteById(groupId);
+    }
+
+    @Test
+    public void testDeleteProductWithoutGroup() {
+        //Given
+        //When
+        group.getProducts().get(0).setProductGroupEntity(null);
```

Comment:
Nie rozumiem tego zapisu. Dlaczego zmieniasz przypisanie do grupy na null?

Reply:
Jedna z grup została już usunięta a w metodzie oznaczonej after jest jest oznaczenie że w przypadku null nie będzie powtórzone usuwanie

Reply:
Jeżeli chodzi o update to po prostu żeby zobaczyć że dokonała się zmiana nie ma innej grupy żeby tam przypisać więc wystarczy null

Reply:
Jeżeli chodzi o update to po prostu żeby zobaczyć że dokonała się zmiana nie ma innej grupy żeby tam przypisać więc wystarczy null. Taka sama zmiana jak nazwy czy ceny

Reply:
Tak, w update rozumiem, ale ten test o który pytam dotyczy usunięcia produktu bez usunięcia grupy.

Reply:
Nie można usunąć tak po prostu elementy jeśli jest on w relacji

Reply:
No właśnie teoretycznie nie. Ale z drugiej strony w aplikacjach jest to rozwiązane tak, że jak np. usuwasz jakiś produkt, to grupa zostaje. Czy to znaczy, że musi być on wcześniej przekazany do innej grupy? Bo widzę, że tutaj jest dodany cascadeType.all i rozumiem, że usunięcie grupy = usunięcie produktów, ale w drugą stronę nie powinno to tak działać teoretycznie.

Reply:
Jeśli produkt nie ma przypisanej grupy ani żadna grupa tego produktu to można go tak po prostu usunąć

Reply:
chyba, że powinniśmy jeszcze się przyjrzeć tym encjom i zdecydować czy ten cascadeType faktycznie taki powinien być.

Reply:
Ale co ci przeszkadza

Reply:
Wiesz co... masz rację. Jakie to ma znaczenie. Ja pracuję przy implementacji sklepów internetowych i chyba po prostu chciałam, żeby to działało dobrze, bo może (skrzywienie? :P). A przecież to ma po prostu działać. I po prostu działa. Nic mi nie przeszkadza.